ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/operations/AttesterSlashing.java

```diff
@@ -31,9 +31,10 @@ public static class AttesterSlashingSchema
       extends ContainerSchema2<AttesterSlashing, IndexedAttestation, IndexedAttestation> {
 
     public AttesterSlashingSchema(
-        final IndexedAttestation.IndexedAttestationSchema indexedAttestationSchema) {
+        final IndexedAttestation.IndexedAttestationSchema indexedAttestationSchema,
+        final boolean isElectra) {
       super(
-          "AttesterSlashing",
+          isElectra ? "AttesterSlashingElectra" : "AttesterSlashingPhase0",
```

Comment:
The problem is that without this change which allows us to differentiate the two (Phase0 and Electra) schemas we can't be aligned with the [API spec](https://github.com/ethereum/beacon-APIs/blob/860584e71109cc9dc34ab6b67c646ae9a818c07e/apis/beacon/pool/attester_slashings.v2.yaml#L25)
Both [`IndexedAttestation`](https://github.com/ethereum/consensus-specs/blob/dev/specs/electra/beacon-chain.md#indexedattestation) and the [`AttesterSlashing`](https://github.com/ethereum/consensus-specs/blob/dev/specs/electra/beacon-chain.md#attesterslashing) containers have been marked as modified in the Electra consensus spec update and that should be reflected in our code (the same way as the [`Attestation`](https://github.com/ethereum/consensus-specs/blob/dev/specs/electra/beacon-chain.md#attesterslashing) change). But since in this case, the change is the list size only, we decided to go with applying the container name and list size changes at runtime, which might look a bit hacky but changing it "properly" would be very expensive (changing the `IndexedAttestation` container implies a huge change cascade up down the `BeaconBlockBody`) and it's not worth the effort.
This is basically impacting an API which is not even public. IMO it's OK to keep it way for now

Reply:
Plus, as you mentioned, we're not even supporting the maxItem constraints at the API level... So unless we get that done, this would be sufficient

Reply:
BTW, I removed the size limit test since it's not really testing the API but rather the backend... We might add such test if we add the `maxItem` constraint